src/app/containers/SettingsContainer.js

```diff
@@ -1,13 +1,25 @@
-import React from 'react';
-import { AppsSidebar, Sidebar, StorageSpaceStatus, Href } from 'react-components';
-import { Route, Switch } from 'react-router';
+import React, { useEffect, useRef } from 'react';
+import { AppsSidebar, Sidebar, StorageSpaceStatus, Href, MainAreaContext, useUser } from 'react-components';
+import { Route, Switch, Redirect } from 'react-router';
 import { c } from 'ttag';
 
 import PrivateHeader from '../content/PrivateHeader';
-import SettingsPage from '../pages/SettingsPage';
+import GeneralPage from '../pages/GeneralPage';
+import ContactGroupsPage from '../pages/ContactGroupsPage';
 
 const SettingsContainer = () => {
-    const list = [{ link: '/contacts/settings', icon: 'contacts', text: c('Link').t`General` }];
+    const mainAreaRef = useRef();
+    const [{ hasPaidMail }] = useUser();
+
+    useEffect(() => {
+        mainAreaRef.current.scrollTop = 0;
+    }, [location.pathname]);
```

Comment:
should we pass `location` as a prop (wrapping `SettingsContainer` with `withRouter`)?

Reply:
Good catch.